diff --git "a/C:\\Users\\andrew\\AppData\\Local\\Temp\\TortoiseGit\\boaBC24.tmp\\board-overo-c5acde1-left.c" "b/E:\\Kernel Development\\linux-omap-revc\\arch\\arm\\mach-omap2\\board-overo.c"
index a381b4f..a6f522f 100644
--- "a/C:\\Users\\andrew\\AppData\\Local\\Temp\\TortoiseGit\\boaBC24.tmp\\board-overo-c5acde1-left.c"
+++ "b/E:\\Kernel Development\\linux-omap-revc\\arch\\arm\\mach-omap2\\board-overo.c"	
@@ -557,18 +557,29 @@ static struct twl4030_platform_data overo_twldata = {
 	.vmmc1		= &overo_vmmc1,
 };
 
-#if defined(CONFIG_RTC_DRV_DS3232) || \
-	defined(CONFIG_RTC_DRV_DS3232_MODULE)
 
+//We support multiple clocks based on Board Rev 
+//Rev C is the DS3231 which is supported by the DS1307 driver
+//Rev B is the DS3232 which is supported by the DS3232 driver
+//So we order based on preference to the newer board
 static struct i2c_board_info __initdata overo_i2c3_boardinfo[] = {
+#if defined(CONFIG_RTC_DRV_DS1307) || \
+	defined(CONFIG_RTC_DRV_DS1307_MODULE)
+	{
+		I2C_BOARD_INFO("rtc-ds1307", 0x68),
+		.type   = "ds3231",
+//		.irq	= 
+	},
+#elif defined(CONFIG_RTC_DRV_DS3232) || \
+	defined(CONFIG_RTC_DRV_DS3232_MODULE)
 	{
 		I2C_BOARD_INFO("rtc-ds3232", 0x68),
 		.type   = "ds3232",
 	},
-};
-#else
-static struct i2c_board_info __initdata overo_i2c3_boardinfo[] = {};
 #endif
+};
+
+
 
 static int __init overo_i2c_init(void)
 {
